src/Controllers/CoursesAPIController.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+
+namespace App\Controllers;
+
+
+use App\Models\CoursesModel;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class CoursesAPIController
+{
+    protected CoursesModel $model;
+
+    // Here, the parameter is automatically supplied by the Dependency Injection Container based on the type hint
+    public function __construct(CoursesModel $model)
+    {
+        $this->model = $model;
+    }
+
+    public function __invoke(RequestInterface $request, ResponseInterface $response): ResponseInterface
```

Comment:
This is one thing to think about.

The ResponseInterface does not include the `withJson` method, which means phpstorm gives some weird warnings when you use it. 

We could fix this by changing the type-hint to `Slim\Http\Response`, but then it looks a bit weird having the response be an interface, but the request being an actual class. 

Does anyone have any thoughts?

Reply:
I have a better idea
https://github.com/slimphp/Slim-Http/pull/194